.github/workflows/tag-and-release.yaml

```diff
@@ -6,8 +6,24 @@ on:
       - main
 
 jobs:
+  
+  # Run standard build tests first
+  run-build-test:
+    uses: ./.github/workflows/build-test.yaml
+    permissions:
+      id-token: write
+      contents: read
+
+  # Run airgap build tests
+  run-airgap-build-test:
+    uses: ./.github/workflows/airgap-build-test.yaml
+    permissions:
+      id-token: write
+      contents: read
+
+  # Only proceed to tag a new version if all tests pass
   tag-new-version:
-    permissions: write-all
```

Comment:
Deletion of permissions was probably just an accident here? CodeQL doesn't like it - https://github.com/defenseunicorns/uds-k3d/security/code-scanning/3